src/batou/lib/file.py

```diff
@@ -513,7 +513,7 @@ def verify(self, predicting=False):
                 output.line(
                     f"see {diff_log} for the full diff.".format(), yellow=True
                 )
-            if contains_secrets:
+            if contains_secrets and self.sensitive_data is None:
```

Comment:
When I looked at this code quickly last week, I thought it would be best to extract the "contains_secrets" part into an earlier part of the if/else chain so that we would only end up with a single `output.annotate` instance for this reason - or at least making it more compact.

Reply:
Okay, I've taken a shot at refactoring the logic to be slightly flatter. I'm not 100% sure if this was what you had in mind, but if not then this might be something to review synchronously at some later date.